src/vmm/src/vstate/system.rs

```diff
@@ -59,10 +59,27 @@ impl KvmContext {
 
         // A list of KVM capabilities we want to check.
         #[cfg(target_arch = "x86_64")]
-        let capabilities = vec![Irqchip, Ioeventfd, Irqfd, UserMemory, SetTssAddr];
+        let capabilities = vec![
+            Irqchip,
+            Ioeventfd,
+            Irqfd,
+            UserMemory,
+            SetTssAddr,
+            Pit2,
+            PitState2,
+            AdjustClock,
+            Debugregs,
+            MpState,
+            VcpuEvents,
+            Xcrs,
+            Xsave,
+            ExtCpuid,
+        ];
 
         #[cfg(target_arch = "aarch64")]
-        let capabilities = vec![Irqchip, Ioeventfd, Irqfd, UserMemory, ArmPsci02];
+        let capabilities = vec![
+            Irqchip, Ioeventfd, Irqfd, UserMemory, ArmPsci02, DeviceCtrl, MpState,
```

Comment:
`OneReg` capability needs to be added for the KVM_GET_ONE_REG and KVM_SET_ONE_REG ioctl.

Reply:
I'll add that. 
Just for my info, where exactly are we using `KVM_GET_ONE_REG`  and `KVM_SET_ONE_REG`?
I see that we are using `KVM_GET_REGS` and `KVM_SET_REGS` via `VcpuFd::get_regs()` and `VcpuFd::set_regs()`
I wasnt able to find the specific API in the wrapper that maps to `KVM_{GET,SET}_ONE_REG`

Reply:
The get_one_reg functionality is the method used for retrieving the value of a register on aarch64. We are using it in multiple places: https://github.com/firecracker-microvm/firecracker/blob/17e259c2be64320a8d2d481813aefdd5f493b47c/src/arch/src/aarch64/regs.rs#L235, https://github.com/firecracker-microvm/firecracker/blob/17e259c2be64320a8d2d481813aefdd5f493b47c/src/arch/src/aarch64/regs.rs#L249. We use set_one_reg: https://github.com/firecracker-microvm/firecracker/blob/17e259c2be64320a8d2d481813aefdd5f493b47c/src/arch/src/aarch64/regs.rs#L385, https://github.com/firecracker-microvm/firecracker/blob/17e259c2be64320a8d2d481813aefdd5f493b47c/src/arch/src/aarch64/regs.rs#L164.

Reply:
Got it, thanks :)
I'm not sure if I was looking in the wrong place, or that these methods are not [documented](https://docs.rs/kvm-ioctls/0.6.0/kvm_ioctls/struct.VcpuFd.html)

I'll push the changes

Reply:
I think the documentation of non- `x86` components is not being rendered on [docs.rs](https://docs.rs/kvm-ioctls/0.6.0/i686-unknown-linux-gnu/kvm_ioctls/struct.VcpuFd.html)

The [build](https://docs.rs/about/builds) documentation on docs.rs states that the default compilation target is  `x86_64-unknown-linux-gnu`, which probably leads to the missing documentation for `aarch64` specific components.

I'll create an issue in the [upstream](https://github.com/rust-vmm/kvm-ioctls) to track this.